test/testautomation_intrinsics.c

```diff
@@ -0,0 +1,717 @@
+/**
+ * Intrinsics test suite
+ */
+#include <SDL3/SDL.h>
+#include <SDL3/SDL_intrin.h>
+#include <SDL3/SDL_test.h>
+#include "testautomation_suites.h"
+
+// FIXME: missing tests for loongarch lsx/lasx
+// FIXME: missing tests for powerpc altivec
+
+/* ================= Test Case Implementation ================== */
+
+/* Helper functions */
+
+static int allocate_random_int_arrays(Sint32 **dest, Sint32 **a, Sint32 **b, size_t *size) {
+    size_t i;
+
+    *size = (size_t)SDLTest_RandomIntegerInRange(127, 999);
+    *dest = SDL_malloc(sizeof(Sint32) * *size);
+    *a = SDL_malloc(sizeof(Sint32) * *size);
+    *b = SDL_malloc(sizeof(Sint32) * *size);
+
+    if (!*dest || !*a || !*b) {
+        SDLTest_AssertCheck(SDL_FALSE, "SDL_malloc failed");
+        return -1;
+    }
+
+    for (i = 0; i < *size; ++i) {
+        (*a)[i] = SDLTest_RandomSint32();
+        (*b)[i] = SDLTest_RandomSint32();
+    }
+    return 0;
+}
+
+static int allocate_random_float_arrays(float **dest, float **a, float **b, size_t *size) {
+    size_t i;
+
+    *size = (size_t)SDLTest_RandomIntegerInRange(127, 999);
+    *dest = SDL_malloc(sizeof(float) * *size);
+    *a = SDL_malloc(sizeof(float) * *size);
+    *b = SDL_malloc(sizeof(float) * *size);
+
+    if (!*dest || !*a || !*b) {
+        SDLTest_AssertCheck(SDL_FALSE, "SDL_malloc failed");
+        return -1;
+    }
+
+    for (i = 0; i < *size; ++i) {
+        (*a)[i] = SDLTest_RandomUnitFloat();
+        (*b)[i] = SDLTest_RandomUnitFloat();
+    }
+
+    return 0;
+}
+
+static int allocate_random_double_arrays(double **dest, double **a, double **b, size_t *size) {
+    size_t i;
+
+    *size = (size_t)SDLTest_RandomIntegerInRange(127, 999);
+    *dest = SDL_malloc(sizeof(double) * *size);
```

Comment:
For the segfault, can you try changing these into `SDL_aligned_alloc(..., 8)`? And higher powers of 2?

Reply:
Will try shortly

Reply:
8 worked for me so far, but the setup I'm running on doesn't support avx or sse4.2:
so, use SDL_SIMDAlignment() for it, maybe?

Reply:
Well, I spoke too soon I guess. It still segs, less frequently though, in `kernel_doubles_add_sse2`

Reply:
Yes, that sounds good.
Weird that alignment caused issues, because I used intrinsics that required no alignment. Perhaps it's mentioned somewhere. What cpu are you testing on?

Reply:
Yes, please do. Then there's no more need for aligned allocation.

Reply:
What value does `SDL_SIMDAlignment` return?

Reply:
> `SDL_SIMDAlignment`

16